lib/csv/table.rb

```diff
@@ -1028,19 +1028,22 @@ def dig(index_or_header, *index_or_headers)
     end
 
     # :call-seq:
-    #   table.inspect => string
+    #   table.inspect(row_counts) -> self
     #
-    # Returns a <tt>US-ASCII</tt>-encoded \String showing table:
-    # - Class: <tt>CSV::Table</tt>.
-    # - Access mode: <tt>:row</tt>, <tt>:col</tt>, or <tt>:col_or_row</tt>.
-    # - Size:  Row count, including the header row.
-    #
-    # Example:
+    # Return table with specified rows data with header
     #   source = "Name,Value\nfoo,0\nbar,1\nbaz,2\n"
     #   table = CSV.parse(source, headers: true)
-    #   table.inspect # => "#<CSV::Table mode:col_or_row row_count:4>"
-    def inspect
-      "#<#{self.class} mode:#{@mode} row_count:#{to_a.size}>".encode("US-ASCII")
+    #   table.inspect(2) # => #<CSV::Row "Name":"foo" "Value":"0">, #<CSV::Row "Name":"bar" "Value":"1">
+    #
+    def inspect(row_counts)
```

Comment:
Generally, `#inspect` should not require any arguments. It breaks compatibility and some tools such as irb can't work with this object.
How about using the constant `row_counts` such as 5 instead of receiving it from argument?